### PR TITLE
Feature: search appointment tests

### PR DESCRIPTION
**Pull request by cakareli**

### CODE REVIEW

File: WPFHospitalEditorUnitTests/SearchAppointmentTest.cs [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+            InitRecommendationDto();
+        }
+
+        [Fact]
+        public void Find_5_appointments_time_priority()
+        {
+
+            StubRepositories();
+            List<RecommendationDto> availableAppointments = new List<RecommendationDto>();
+
```
Comment (rammba) [inline WPFHospitalEditorUnitTests/SearchAppointmentTest.cs:67]: Do you use availableAppointments in any test?

File: WPFHospitalEditorUnitTests/SearchAppointmentTest.cs [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+        }
+
+        [Fact]
+        public void Find_0_appointments_with_doctor_priority()
+        {
+            StubRepositories();
+            List<RecommendationDto> availableAppointments = new List<RecommendationDto>();
+
+            examinationService.Recommend(dto6).Count.ShouldBe(0);
+        }
```
Comment (BoKna031) [inline WPFHospitalEditorUnitTests/SearchAppointmentTest.cs:114]: All test are identical. They are different only in the dto data that is sent and in the expected response. Use [THEORY ](https://www.youtube.com/watch?v=SqI1BfYg0No&feature=youtu.be&ab_channel=NikolaLuburi%C4%87)to avoid unnecessary code copying.

File: WPFHospitalEditorUnitTests/SearchAppointmentTest.cs [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+        private Shift shift5;
+        private RecommendationRequestDto dto1;
+        private RecommendationRequestDto dto2;
+        private RecommendationRequestDto dto3;
+        private RecommendationRequestDto dto4;
+        private RecommendationRequestDto dto5;
+        private RecommendationRequestDto dto6;
+        private List<Shift> shifts1 = new List<Shift>();
+        private List<Shift> shifts2 = new List<Shift>();
+        private List<Shift> shifts3 = new List<Shift>();
```
Comment (BoKna031) [inline WPFHospitalEditorUnitTests/SearchAppointmentTest.cs:46]: You store too much unnecessary data as  attributes.

**Review by BoKna031 — COMMENTED** [review]
Use theory to remove similar tests and remove or reduce number of attributes.

File: WPFHospitalEditorUnitTests/SearchAppointmentTest.cs (hunk 1 above)
Comment (cakareli) [reply WPFHospitalEditorUnitTests/SearchAppointmentTest.cs]: No, I deleted it.

File: WPFHospitalEditorUnitTests/SearchAppointmentTest.cs (hunk 2 above)
Comment (cakareli) [reply WPFHospitalEditorUnitTests/SearchAppointmentTest.cs]: Fixed

File: WPFHospitalEditorUnitTests/SearchAppointmentTest.cs (hunk 3 above)
Comment (cakareli) [reply WPFHospitalEditorUnitTests/SearchAppointmentTest.cs]: Only doctor, timeInterval and room attributes are left.

File: WPFHospitalEditorUnitTests/RecommendAppointmentsTest.cs [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+                Preference = RecommendationPreference.Doctor
+            }, 5 });
+
+            retVal.Add(new object[] { new RecommendationRequestDto()
+            {
+                DoctorId = doctors[2].Id,
+                TimeInterval = timeIntervals[1],
+                SpecialtyId = 1,
+                Preference = RecommendationPreference.Doctor
+            }, 3 });
```
Comment (rammba) [inline WPFHospitalEditorUnitTests/RecommendAppointmentsTest.cs:105]: I think it's enough to test case with zero and case with multiple appointments. No need to test three and five available appointments.

File: WPFHospitalEditorUnitTests/RecommendAppointmentsTest.cs [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+{
+    public class RecommendAppointmentsTest
+    {
+        private readonly Mock<IShiftRepository> mockShiftRepository = new Mock<IShiftRepository>();
+        private readonly Mock<IExaminationRepository> mockExaminationRepository = new Mock<IExaminationRepository>();
+        private readonly Mock<IDoctorRepository> mockDoctorRepository = new Mock<IDoctorRepository>();
+        private readonly ExaminationService examinationService;
+        private static Dictionary<int, Doctor> doctors =  new Dictionary<int, Doctor>();
+        private static List<TimeInterval> timeIntervals =  new List<TimeInterval>();
+        private static Dictionary<int, Room> rooms = new Dictionary<int, Room>();
```
Comment (BoKna031) [inline WPFHospitalEditorUnitTests/RecommendAppointmentsTest.cs:29]: There is no need for the variables to be static

File: WPFHospitalEditorUnitTests/RecommendAppointmentsTest.cs (hunk 4 above)
Comment (cakareli) [reply WPFHospitalEditorUnitTests/RecommendAppointmentsTest.cs:105]: Five available appointments is maximum amount of displayed appointments. So I tested when there is zero and five available appointments(minimum and maximum). Also I wanted to test if there is number of appointments between minimum and maximum, and because of that I tested case with three appointments.

File: WPFHospitalEditorUnitTests/RecommendAppointmentsTest.cs (hunk 5 above)
Comment (cakareli) [reply WPFHospitalEditorUnitTests/RecommendAppointmentsTest.cs:29]: These variables must be static because they are used in Recommendation_request_data method which is also static.

**Review by rammba — APPROVED** [review]

**Comment by rammba** [comment]
[(0) PR Feature: search appointment tests](https://trello.com/c/cC9RKjzG/120-0-pr-feature-search-appointment-tests)

**Review by BoKna031 — APPROVED** [review]
